Devscord.DiscordFramework/Framework/Commands/AntiSpam/Models/ServerMessagesCacheService.cs

```diff
@@ -62,7 +62,7 @@ private static void RemoveOldMessagesCyclic()
             {
                 list.RemoveAll(message => message.SentAt < minTimeInPast);
                 return list;
-            }).ToList();
+            }).Where(x => x.Any()).ToList();
             if (smallMessages.Count == 0)
             {
                 return;
```

Comment:
po co te ToList? Wywal to oraz zamień Count == 0 na !Any.

Reply:
po ta by uniknąć multiple enumeration
Gdyby nie było ToList to całość byłaby liczona dwa razy
a co do Any to zmienię